File: BeeFit.API/Models/Dish.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BeeFit.API.Models
{
    public class Dish
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [ForeignKey("UserId")]
        public virtual User User { get; set; }
        public int UserId { get; set; }
        public float Callories { get; set; }
        public float Fats { get; set; }
        public float Proteins { get; set; }
        public float Carbohydrates { get; set; }
        public virtual ICollection<DishesIngredient> Ingredients { get; set; }
    }
}

File: BeeFit-SPA/src/app/_models/Dish.ts
import { User } from './User';
import { DishesIngredient } from './DishesIngredient';

export class Dish {
    id: number;
    name: string;
    userId: number;
    ingredients: DishesIngredient[];
    callories: number;
    fats: number;
    carbohydrates: number;
    proteins: number;
    weight?: number;
}
